src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1867,7 +1867,8 @@ MulticopterPositionControl::task_main()
 					/* limit min lift */
 					if (-thrust_sp(2) < thr_min) {
 						thrust_sp(2) = -thr_min;
-						saturation_z = true;
+						/* Don't freeze altitude integral if it wants to throttle up */
+						saturation_z = vel_err(2) > 0.0f ? true : saturation_z;
 					}
```

Comment:
We are here when vertical thrust setpoint demanded is less than minimum. We do not want to saturate integrals if they what to increase the vertical thrust setpoint above this minimum value.